### PR TITLE
#167751571 improve the backends

### PR DESCRIPTION
**Pull request by NiyongaboEric**
### What does this PR do?

Way-Farer backend contains the major improvement from the existing one the API documentation, Authentication, endpoints, and tests files modified.

### Description of task completed
**Expected:**

*ROOT FOLDER*
- .env should not appear on Github
- .coveralls.yml should not appear on GitHub
- app.js should not have comment and spaces
- app.js no longer use body-parser modules
- server.js files comments are removed

*CONTROLLERs*
 *Users*
- the variable declaration is small
- user controller has no comment
- all controller files are in came case with the controller
- ids generation is no longer based on the length of the array
- ids are no longer duplicates
- the variable is destructed using es6
- control name should be in camel case

*Trips*
-the trip class is tripController class
- the id does not base on length of the array
- cancel trip methods should destruct data

*Bookings*
- destructure variable using ES6 destruction
- JSDOC is added on top of classes
- trip condition is checking before user finding
- the token payload is improved

*ROUTEs*
 - main.js file name changed
- middleware use ES6 modules

*HELPERs*
- use es6 modules
- removed unwanted class methods

### What are the relevant pivotal tracker stories? 
[#167751571](https://www.pivotaltracker.com/story/show/167751571)

### CODE REVIEW

**Comment by NiyongaboEric** [comment]
create_trip
local_coveralls
local_heroku
signin
static_ui

File: .env.example [hunk 1]
```diff
@@ -0,0 +1 @@
+SECRET_KEY=''
```
Comment (oesukam) [inline .env.example]: Blank line

File: .eslintrc.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
         "sourceType": "module"
     },
     "rules": {
-        "class-methods-use-this": 0
+        "class-methods-use-this": 0,
+        "no-console": 0
     }
 }
```
Comment (oesukam) [inline .eslintrc.json]: Empty line required

File: .gitignore [hunk 3]
```diff
@@ -2,4 +2,6 @@ node_modules/
 package-lock.json
```
Comment (oesukam) [inline .gitignore]: Blank line

File: .gitignore [hunk 4]
```diff
@@ -2,4 +2,6 @@ node_modules/
 package-lock.json
 
 .nyc_output
-coverage
\ No newline at end of file
+coverage
+
+.env
```
Comment (oesukam) [inline .gitignore]: Empty line required

**Pull request merged**